### PR TITLE
feat(gtk): support create Ed25519 in gtk

### LINKED ISSUE #1486: Support Generation of Ed25519 Address in GUI
## Describe the problem you'd like to have solved

The current GUI does not support the generation of Ed25519 addresses, which limits the usability of Ed25519 in the application. Adding this feature will allow users to generate Ed25519 addresses directly from the GUI, improving the user experience and broadening the tool's capabilities.

- [x] [feat(wallet): supporting Ed25519 curve in wallet](https://github.com/pactus-project/pactus/pull/1484)

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

This PR support generation Ed25519 account type on GTK.

image
image

## Related issue(s)

- Fixes #1486

### CODE REVIEW

**Comment by b00f** [comment]
Why don't you get password before generating the Ed25519?
This way we keep the dialog as i t is and get password once needed.,

**Comment by Ja7ad** [comment]
> Why don't you get password before generating the Ed25519? This way we keep the dialog as i t is and get password once needed.,

fixed

File: cmd/gtk/dialog_wallet_create_address.go [hunk 1]
```diff
@@ -17,31 +16,43 @@ func createAddress(ww *widgetWallet) {
...
 		walletAddressLabel, err := addressLabel.GetText()
 		fatalErrorCheck(err)
 
 		walletAddressType := addressTypeCombo.GetActiveID()
 		fatalErrorCheck(err)
 
-		if walletAddressType == wallet.AddressTypeBLSAccount {
+		switch walletAddressType {
+		case wallet.AddressTypeEd25519Account:
+			_, err = ww.model.wallet.NewEd25519AccountAddress(walletAddressLabel, password)
```
Comment (b00f) [inline cmd/gtk/dialog_wallet_create_address.go:51]: Here you can show the password dialog. No need to create the password fetcher callback

**Comment by b00f** [comment]
@Ja7ad please update the description

File: cmd/gtk/dialog_wallet_create_address.go [hunk 2]
```diff
@@ -21,27 +20,39 @@ func createAddress(ww *widgetWallet) {
...
+		case wallet.AddressTypeEd25519Account:
+			passwordFetcher := func(wlt *wallet.Wallet) (string, bool) {
+				if *passwordOpt != "" {
+					return *passwordOpt, true
+				}
+
+				return getWalletPassword(wlt)
+			}
+
+			password, ok := passwordFetcher(ww.model.wallet)
```
Comment (b00f) [inline cmd/gtk/dialog_wallet_create_address.go:46]: @Ja7ad why don't we use it like:
```go
password, ok := getWalletPassword(wlt)
		if ok {
			_, err = ww.model.wallet.NewEd25519AccountAddress(walletAddressLabel, password)
		}
```

File: cmd/gtk/dialog_wallet_create_address.go [hunk 3]
```diff
@@ -21,27 +20,39 @@ func createAddress(ww *widgetWallet) {
...
 
 	addressTypeCombo := getComboBoxTextObj(builder, "id_combo_address_type")
-	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "Account")
+	addressTypeCombo.Append(wallet.AddressTypeEd25519Account, "ED25519 Account")
+	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "BLS Account")
 	addressTypeCombo.Append(wallet.AddressTypeValidator, "Validator")
 
 	addressTypeCombo.SetActive(0)
 
-	getButtonObj(builder, "id_button_ok").SetImage(OkIcon())
```
Comment (b00f) [inline cmd/gtk/dialog_wallet_create_address.go:29]: Why do you remove these code?

File: cmd/gtk/dialog_wallet_create_address.go (hunk 2 above)
Comment (Ja7ad) [reply cmd/gtk/dialog_wallet_create_address.go:46]: Not good, ED is default type and always open create account modal show password.

Comment (b00f) [reply cmd/gtk/dialog_wallet_create_address.go:46]: What do you mean default type?

File: cmd/gtk/dialog_wallet_create_address.go [hunk 4]
```diff
@@ -21,13 +20,13 @@ func createAddress(ww *widgetWallet) {
...
 	addressTypeCombo := getComboBoxTextObj(builder, "id_combo_address_type")
-	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "Account")
+	addressTypeCombo.Append(wallet.AddressTypeEd25519Account, "ED25519 Account")
+	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "BLS Account")
 	addressTypeCombo.Append(wallet.AddressTypeValidator, "Validator")
 
 	addressTypeCombo.SetActive(0)
 
 	getButtonObj(builder, "id_button_ok").SetImage(OkIcon())
-	getButtonObj(builder, "id_button_cancel").SetImage(CancelIcon())
```
Comment (b00f) [inline cmd/gtk/dialog_wallet_create_address.go:30]: Why this line is removed?

File: cmd/gtk/dialog_wallet_create_address.go [hunk 5]
```diff
@@ -21,13 +20,13 @@ func createAddress(ww *widgetWallet) {
 	addressLabel := getEntryObj(builder, "id_entry_account_label")
 
 	addressTypeCombo := getComboBoxTextObj(builder, "id_combo_address_type")
-	addressTypeCombo.Append(wallet.AddressTypeBLSAccount, "Account")
+	addressTypeCombo.Append(wallet.AddressTypeEd25519Account, "ED25519 Account")
```
Comment (b00f) [inline cmd/gtk/dialog_wallet_create_address.go:23]: We can define String function for Address Types

**Review by b00f — CHANGES_REQUESTED** [review]

File: cmd/gtk/dialog_wallet_create_address.go (hunk 5 above)
Comment (Ja7ad) [reply cmd/gtk/dialog_wallet_create_address.go:23]: Address types in wallet is not enum and don't have specific type for define String method.

We can use crypto.AddressType in future.

**Review by b00f — APPROVED** [review]

**Pull request merged**